File: src/YowThi.Erp.Application/Labor/LaborDailyWageOptions.cs
namespace YowThi.Erp.Application.Labor;

public sealed record LaborDailyWageOptionsQuery(string Locale, string? Search, int Offset, int Limit);
public sealed record LaborDailyWageOptionPage<T>(IReadOnlyList<T> Items, int? NextOffset);
public sealed record LaborDailyWageEmployeeOption(Guid Id, string DisplayName);
public sealed record LaborProcessingWageTargetOption(Guid ProcessingModuleOutputId, string DisplayName, decimal ConfiguredWageRateSnapshot, decimal AggregatedQuantity);
public sealed record LaborDailyWageWorkspace(Guid EmployeeId, DateOnly WorkDate, string EmployeeDisplayName, bool AlreadyConfirmed, IReadOnlyList<LaborProcessingWageTargetOption> ProcessingTargets, int SalesPackagingWorkRecordCount, long SalesPackagingWageTotalThb);

public interface ILaborDailyWageOptionsReader
{
    ValueTask<LaborDailyWageOptionPage<LaborDailyWageEmployeeOption>> GetEmployeesAsync(LaborDailyWageOptionsQuery query, CancellationToken cancellationToken);
    ValueTask<LaborDailyWageWorkspace?> GetWorkspaceAsync(Guid employeeId, DateOnly workDate, string locale, CancellationToken cancellationToken);
}

File: src/YowThi.Erp.Web/src/features/labor/laborDailyWageOptions.ts
import { getApiJson } from '../../app/api/apiTransport';
import type { OperationalLocale } from '../../app/i18n/locale';

export interface LaborDailyWageEmployeeOption { id: string; displayName: string; }
export interface LaborDailyWageEmployeeOptionsResponse { items: LaborDailyWageEmployeeOption[]; nextCursor: string | null; }
export interface LaborProcessingWageTarget { processingModuleOutputId: string; displayName: string; configuredWageRateSnapshot: number; aggregatedQuantity: number; }
export interface LaborDailyWageWorkspace { employeeId: string; workDate: string; employeeDisplayName: string; alreadyConfirmed: boolean; processingTargets: LaborProcessingWageTarget[]; salesPackagingWorkRecordCount: number; salesPackagingWageTotalThb: number; }

interface EmployeeOptionsQuery { locale: OperationalLocale; search?: string; cursor?: string | null; limit?: number; signal?: AbortSignal; }

export async function listLaborDailyWageEmployeeOptions(query: EmployeeOptionsQuery): Promise<LaborDailyWageEmployeeOptionsResponse> {
  const parameters = new URLSearchParams();
  const search = query.search?.trim();
  if (search) parameters.set('search', search);
  if (query.cursor) parameters.set('cursor', query.cursor);
  if (query.limit !== undefined) parameters.set('limit', String(query.limit));
  const suffix = parameters.size === 0 ? '' : `?${parameters.toString()}`;
  return getApiJson<LaborDailyWageEmployeeOptionsResponse>(`/api/v1/labor/daily-wage-options/employees${suffix}`, { locale: query.locale, signal: query.signal });
}

export async function getLaborDailyWageWorkspace(employeeId: string, workDate: string, locale: OperationalLocale, signal?: AbortSignal): Promise<LaborDailyWageWorkspace> {
  const parameters = new URLSearchParams({ workDate });
  return getApiJson<LaborDailyWageWorkspace>(`/api/v1/labor/employees/${employeeId}/daily-wage-workspace?${parameters.toString()}`, { locale, signal });
}
